### PR TITLE
REL-1053: Add conn_backlog parameter

### PR DESCRIPTION
**Pull request by Zenebatos**
This PR performs the following:

- Adds the conn_backlog parameter
- Converts the module to pdk, taking care to configure it for puppet 3.8.7
- Fixes all rubocop validation failures

### CODE REVIEW

File: templates/memcached_windows.erb [hunk 1]
```diff
@@ -28,6 +28,9 @@ result << '-t ' + @processorcount
 if @max_connections 
   result << '-c ' + @max_connections
 end
+if @conn_backlog
+  resulot << "-b #{@conn_backlog.to_s}"
```
Comment (bvujanic) [inline templates/memcached_windows.erb]: s/resulot/result?

**Review by bvujanic — COMMENTED** [review]
Besides the misspell everything looks good to me.

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -82,7 +82,8 @@
      processorcount: 3,
      use_sasl: true,
      large_mem_pages: true,
-     extended_opts: %w[lru_crawler lru_maintainer]
+     conn_backlog: 500,
```
Comment (bvujanic) [inline spec/classes/init_spec.rb:85]: I think that the default here should be 1024, according to man pages:
```
       -b <num>
              Set the backlog queue limit to <num> connections. The default is 1024.
```

Comment (Zenebatos) [reply spec/classes/init_spec.rb:85]: This is just a test case, not defaults. Defaults are in init.pp. Its defaulting to undef to retain backwards-compatibility w/ the interface.

File: templates/memcached_windows.erb (hunk 1 above)
Comment (Zenebatos) [reply templates/memcached_windows.erb]: Fixed

**Comment by Zenebatos** [comment]
Added max_sequential argument for -R option.

**Review by bvujanic — APPROVED** [review]

**Pull request merged**